### DISCUSSION TITLE
Option for some parts to opt out of the 'change detection' system

### DISCUSSION THREAD

**Discussion by bossanova808** [craftcms/cms#13936] [Ideas] [answered]
We use Ben's excellent Campaign plugin, and this has brought to light a UX issue.  I think this is likely to become more of an issue as more sophisticated back end tools are written, and integrated into the Craft control panel.

As I understand it, at the moment, it's not possible for parts of a page to 'opt out' of the change management system.  That is, if _any_ part of the page within the form is changed, in any way, you get a message 'Showing Unsaved Changes' at the top.  In the case of this scenario where we're just using a bit of the sidebar UI to send test emails, but no actual changes to content are being made, this causes editor confusion (compounded with the fact that  I don't think the multi-user editing side of things is still that great - those little 'user in element' circles are easy to miss, and sometimes also don't seem to update appropriately/speedily).   

In general, editor confusion about the sate of things being saved/not saved is a real pain point as it makes them feel far less confident about this content creation/editing process.

Ideally, there would be a change (I presume around here) - allowing elements (e.g. those with a certain class, I guess?) -  to opt out of the system - i.e. be effectively hidden from the change detection:
https://github.com/craftcms/cms/blob/b8f30d07cd487c5fbaa140743816e6eb7020e55d/src/web/assets/cp/src/js/ElementEditor.js#L1164C11-L1171

(See issue on Ben's plugin here: https://github.com/putyourlightson/craft-campaign/issues/437)

**Comment by brandonkelly** [comment, marked as answer]
If the form element doesn’t have any effect on the campaign element itself, it should just not have a `name` property, in which case it will be excluded from Craft’s form change detection.
